gdx/src/com/badlogic/gdx/utils/Json.java

```diff
@@ -572,6 +572,26 @@ public void writeValue (Object value, Class knownType, Class elementType) {
 				writeObjectEnd();
 				return;
 			}
+			if (value instanceof ObjectIntMap) {
+				if (knownType == null) knownType = ObjectIntMap.class;
+				writeObjectStart(actualType, knownType);
+				for (ObjectIntMap.Entry entry : ((ObjectIntMap<?>)value).entries()) {
+					writer.name(convertToString(entry.key));
```

Comment:
Is there an actual need to serialize maps with numeric keys to JSON? Keys are always strings in JSON and I've not seen the need for this. Note serialization can be customized via the Serializer interface.

Reply:
@NathanSweet `convertToString(entry.key)` aren't they converted to string?

Reply:
Indeed, @Monarchis is right, they are converted to and from String to store in JSON format. Just because you haven't seen the need yet doesn't mean you won't encounter some such need in the future. Passing around Json objects with custom serializers set is a hassle that the library can't do, while IIRC using Json.Serializable required an inner map inside each map, and adapting Json seemed to match the other special treatment that libGDX data structures get in that class.

Reply:
Sorry, I meant to post my comment below, where IntMap and LongMap are handled. There's two issues: 1) non-string keys are not a great fit for JSON data, and 2) the instanceof checks were never intended to be expanded to a long list. It would be better to write various `Json.Serializer`s for each libgdx class we want to support, chosen based on expected need, and those can be registered normally, via `setSerializer` (whether automatically, or by the user one-by-one, or by the user via a new method `registerDefaultSerializers`).

> Passing around Json objects with custom serializers set is a hassle that the library can't do

What does this mean? Configuring the Json instance is standard API usage.

> while IIRC using Json.Serializable required an inner map inside each map

It would be done like this, there is no extra map:
```
Json json = new Json();
json.setOutputType(OutputType.json);
json.setSerializer(IntMap.class, new Serializer<IntMap>() {
	public void write (Json json, IntMap map, Class knownType) {
		json.writeObjectStart();
		for (IntMap.Entry entry : ((IntMap<?>)map).entries())
			json.writeValue(Integer.toString(entry.key), entry.value);
		json.writeObjectEnd();
	}

	public IntMap read (Json json, JsonValue jsonData, Class type) {
		IntMap map = new IntMap(jsonData.size);
		for (JsonValue child = jsonData.child; child != null; child = child.next)
			map.put(Integer.parseInt(child.name), json.readValue(null, child));
		return map;
	}
});
IntMap map = new IntMap();
map.put(1, "one");
map.put(2, "two");
map.put(3, "three");
System.out.println(map); // [3=three, 2=two, 1=one]
String string = json.toJson(map);
System.out.println(string); // {"1":"one","2":"two","3":"three"}
IntMap map2 = json.fromJson(IntMap.class, string);
System.out.println(map2); // [3=three, 2=two, 1=one]
```

Anyway, changes to Json can be separate from this PR, which is already large.